### ISSUE TITLE
Output Binding Always Passes NULL for Columns Not Defined

### ISSUE THREAD

**Issue by chlafreniere** [Azure/azure-functions-sql-extension#75]
Let's take the following simple example, where one column has a default value:

```
-- Create the table in the specified schema
CREATE TABLE [dbo].[Products2]
(
    [ProductID] INT PRIMARY KEY, -- Primary Key column
    [Blah] NVARCHAR(2) NOT NULL DEFAULT 'hi'
);
GO
```

When I run the following query, everything inserts just fine:

```
-- Insert rows into table 'Products2' in schema '[dbo]'
INSERT INTO [dbo].[Products2]
( -- Columns to insert data into
 [ProductID]
)
VALUES
( 
 7
)
-- Add more rows here
GO
```

Using edit data, I can graphically see that data is inserted, and the column 'Blah' has the default value (which is expected):

![image](https://user-images.githubusercontent.com/40371649/134583411-7d649ffb-b317-4bed-ac12-7e21dbcdc6f4.png)

**Now**, let's get this repo in the picture. I used the following sample: https://github.com/Azure/azure-functions-sql-extension/blob/chlafren/fixDatetime/samples/SqlExtensionSamples/OutputBindingSamples/AddProduct.cs. Note that the 'Blah' column isn't part of this sample.

(note that the table has the same primary key column name out of convenience)

When I try to execute this function, I see the following in the console:

```
[2021-09-23T20:58:16.288Z] Executed 'AddProduct' (Failed, Id=3144545e-da87-4788-b466-8697fd19563e, Duration=511ms)
[2021-09-23T20:58:16.288Z] System.Private.CoreLib: Exception while executing function: AddProduct. Microsoft.Azure.WebJobs.Host: Error while handling parameter product after function returned:. Core Microsoft SqlClient Data Provider: Cannot insert the value NULL into column 'Blah', table 'foo.dbo.Products2'; column does not allow nulls. UPDATE fails.
```

The `Cannot insert the value NULL into column 'Blah'` part is key.

I would posit that we shouldn't try to insert `NULL` data for columns that aren't explicitly defined. I'm sure there are other cases beyond default constraints where this behavior can be troublesome.

**Comment by chlafreniere** [comment]
Put the P0 label in to make sure that we chat about this behavior.

**Comment by dzsquared** [comment]
I agree.  To restate what @chlafreniere mentioned - if a column in the AF object is undefined, the query should not upsert against that column.

Bonus points if we validate column properties in the binding and throw an error before hitting the database.  (not null and no default - catch)

**Comment by chlafreniere** [comment]
For the bonus points portion, there is a `COLUMN_DEFAULT` column in `INFORMATION_SCHEMA.COLUMNS` (which we [already query](https://github.com/Azure/azure-functions-sql-extension/blob/fd2445bb8c5262d27a96a7224ff0d01b5655e41f/src/SqlBinding/SqlAsyncCollector.cs#L273)).
